### DISCUSSION TITLE
Best way to extract the trpc/react types?

### DISCUSSION THREAD

**Discussion by aaronvg** [trpc/trpc#1798] [Q&A] [answered]
Our app gets data via other means (Websocket) that contain objects of the same type as those returned from the TRPC client. What's the best way to extract these types used from useQuery(...) etc?

Awesome framework btw!

**Comment by jlalmes** [comment]
tRPC already supports websockets! You can read more here: https://trpc.io/docs/subscriptions

**Reply by aaronvg** [reply]
Ahh right, I'm using Firestore realtime updates on the client which returns an object the shape of a trpc api. But maybe I can create an adapter of sorts or just straight up export the zod schema that matches the trpc response from the server.

**Reply by aaronvg** [reply]
Ah I found the answer, https://trpc.io/docs/infer-types 

thanks!

**Comment by aaronvg** [comment, marked as answer]
The answer is this https://trpc.io/docs/infer-types

**Reply by excal1bu7** [reply]
What about inferring from subscriptions?
